qemu/tests/expose_host_mtu.py

```diff
@@ -0,0 +1,101 @@
+            return 1
+        else:
+            return 0
+
+    def cleanup_ovs_ports(netdst, ports):
+        if is_ovs_backend(netdst) == 1:
+            logging.info("before delete, ovs ports are: %s" %
+                         get_ovs_ports(netdst))
+            logging.info("before vm boot, ovs ports are: %s" %
+                         ports)
+            ports = set(get_ovs_ports(netdst).splitlines()) - \
+                set(ports.splitlines())
+            logging.info("ports need to be deleted: %s" % ports)
+            for p in ports:
+                process.system("ovs-vsctl del-port %s %s" % (netdst, p))
+
+    netdst = params.get("netdst", "switch")
+    if netdst in utils_net.Bridge().list_br():
+        cmd = params["bridge_get_if_cmd"]
+        host_hw_interface = process.system_output(
+                cmd, shell=True).decode().strip(':')
+    elif is_ovs_backend(netdst) == 1:
+        host_hw_interface = get_ovs_ports(netdst)
+    else:
+        test.cancel("The host is using Macvtap backend, which is not"
+                    " supported by now!")
+
+    ports = get_ovs_ports(netdst)
+    params["start_vm"] = "yes"
+    env_process.preprocess_vm(test, params, env, params["main_vm"])
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+
+    device_mac = vm.virtnet[0].mac
+    vm_iface = vm.get_ifname()
+    # TBD, will support windows later
+    process.system_output(params["nm_stop_cmd"])
+    process.system_output(params["set_guest_mtu_cmd"] % host_hw_interface)
+    process.system_output(params["set_guest_mtu_cmd"] % vm_iface)
+
+    os_type = params.get("os_type", "linux")
+    login_timeout = float(params.get("login_timeout", 360))
+    session = vm.wait_for_login(timeout=login_timeout)
+
+    host_ip = utils_net.get_ip_address_by_interface(params["netdst"])
+    if os_type == "linux":  # TBD, will support windows later
+        session.cmd_output_safe(params["nm_stop_cmd"])
+        guest_ifname = utils_net.get_linux_ifname(session,
+                                                  vm.get_mac_address())
+        output = session.cmd_output_safe(
+                params["check_guest_mtu_cmd"] % guest_ifname)
+        error_context.context(output, logging.info)
+        match_string = "mtu %s" % params["mtu_value"]
+        if match_string in output:
+            logging.info("Host mtu %s exposed to guest as expected!" %
+                         params["mtu_value"])
+            logging.info("Ping from guest to host with packet size 3972")
+            status, output = utils_test.ping(host_ip, 10, packetsize=3972,
+                                             timeout=30, session=session)
+            ratio = utils_test.get_loss_ratio(output)
+            if ratio != 0:
+                test.fail("Loss ratio is %s", ratio)
+        else:
+            test.fail("host mtu %s not exposed to guest" % params["mtu_value"])
+        cleanup_ovs_ports(netdst, ports)
+        session.close()
```

Comment:
Better move `cleanup_ovs_ports()` and `session.close()` out of the `if` block, since the action in pairs is out of the `if` condition and has nothing to do with the guest os type.

Reply:
@vivianQizhu Updated.